.github/workflows/codeql.yml

```diff
@@ -0,0 +1,42 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: ["main"]
+    paths-ignore:
+      - "**.md"
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: ["main"]
+    paths-ignore:
+      - "**.md"
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      matrix:
+        # Add the language(s) you want to analyze here as an array of strings
+        # for example: ['javascript'] or ['python', 'javascript']
+        language: ["TODO"]
```

Comment:
I notice CodeQL files because it does not recognize `TODO`. Perhaps we should do this as a comment or give these a different file extension.

Reply:
done in 1797f8828acb2eed20a861f8df7ccbf51169c07d